File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Types/Delegates.vb
Public Delegate Sub WithoutParameters()

Public Delegate Sub WithOneParameter(foo As Int32)

Public Delegate Sub WithMultipleParameters(foo As Int32, bar As Single)

Public Delegate Sub WithDefaultParameters(Optional foo As Int32 = 1)

Public Delegate Sub WithParametersModifiers(ByRef foo As Int32, ByVal bar As Single)

Public Delegate Sub WithGenericParameters(Of T)(foo As T, bar As T())

Public Delegate Function WithReturnType() As Int32

Public Delegate Function WithGenericReturnType(Of T)() As T

Public Delegate Sub SingleGenericType(Of T)()

Public Delegate Sub MultipleGenericType(Of T As {Class, IList(Of String)}, Y As Structure)()

Public Delegate Sub CovariantGenericType(Of In T)()

Public Delegate Sub ContravariantGenericType(Of Out T)()

File: src/Services/Roslyn/Markify.Services.Roslyn.Fixtures/Projects/Source/Types/Delegates.cs
public delegate void WithoutParameters();

public delegate void WithOneParameter(Int32 foo);

public delegate void WithMultipleParameters(Int32 foo, Single bar);

public delegate void WithDefaultParameters(Int32 foo = 1);

public delegate void WithParametersModifiers(ref Int32 foo, out Int32 bar);

public delegate void WithGenericParameters<T>(T foo, T[] bar);

public delegate Int32 WithReturnType();

public delegate T WithGenericReturnType<T>();

public delegate void SingleGenericType<T>();

public delegate void MultipleGenericType<T, Y>()
    where T : class, IList<string>
    where Y : struct;

public delegate void CovariantGenericType<in T>();

public delegate void ContravariantGenericType<out T>();